docs/docs/general/03-Element Selection/recommended-practices.md

```diff
@@ -4,168 +4,242 @@ title: Recommended Practices
+### Coherent Text is Detected as Separate Text-Elements
+Sometimes coherent text is split up into two or more text-elements.
+
+* If you need to match the exact-text: Use a custom element
+* If you just need to interact with the text and it is not important where it is exactly: Target any part of the text
 
 ```typescript
-await askui.click().customElement({customImage: 'custom_elements/red.text.png'})
-                   .or()
-                   .customElement({customImage: 'custom_elements/blue.text.png'}).exec()
+// Custom element instead of text(<your_text>)
+await aui.click().customElement({
+                     customImage: 'custom_elements/mytext.text.png'
+                   })
+                   .exec();
+
+/**
+ *  Given this text will be split in two text elements:
+ *  1lKBASDF_Aeb567878
+ *  First element: 1lKBASDF
+ *  Second element: Aeb567878
+ *
+ *  Target the first element
+ */
+await aui.click().text('1lKBASDF').exec();
 ```
 
-## Recommendation for Fast Speed of Inference
+---
+
+## Detection Fails on a Different Machine / in Pipeline
+When the resolution changes for a workflow run on a different machine or inside a Continuous integration pipeline for example, you may encounter the following problems:
+
+<!-- vale off -->
+* Text linebreaks and text detection fails: See [Text Detection Fails Because of Linebreaks](#text-detection-fails-because-of-linebreaks)
+* Elements change shape or color during a workflow run: See [Color of Same Element Changes](#color-of-same-element-changes) and [Duplicate Elements](#duplicate-elements)
+<!-- vale oon -->
 
-### Avoid Optical Character Recognition (OCR) on Too Many Element
+---
 
-If you use `containsText()`, `withText()`, `withExactText()` or `withTextRegex()` **OCR** is applied to all elements detected on your screen. This can slow down AskUI. It is more efficient to narrow down the elements first. For example, if you want to click a `button` with a specific text you should select all buttons first.
+## Detection Is Flaky or Mislabeled
 
-```javascript
-// Do this
-await aui.click().button('See here').exec();
-await aui.clickText('Sign in');
+### Flaky Detection Across Runs
+Visual detection of elements is highly sensitive to changes in your UI, even if they are small. For example a button that you target may fail randomly on a workflow run every now and then:
 
-// And NOT this
-await aui.click('See here').exec();
-await aui.click('Sign in').exec();
+```typescript
+await aui.click().button().withText('Sign in').exec();
 ```
 
-### Avoid Custom Element Detection if Possible
+You can do three things about this:
 
-If you use `customElement()` you are doing an image-in-image search. Use this sparingly for elements that you can not detect as the execution time is sometimes slower!
+1. Make the element-description more specific, for example with [relational selectors](../../api/04-Relations/above.md)
+2. Add a custom element with `or()` as a fallback
```

Comment:
I think we should always link to custom element and the relation like you did it about for `Relations`. Saw this a couple of times in the text.